textdb/textdb-web/src/main/java/edu/uci/ics/textdb/web/resource/PlanStoreResource.java

```diff
@@ -0,0 +1,140 @@
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import edu.uci.ics.textdb.api.common.ITuple;
+import edu.uci.ics.textdb.api.exception.TextDBException;
+import edu.uci.ics.textdb.api.storage.IDataReader;
+import edu.uci.ics.textdb.planstore.PlanStore;
+import edu.uci.ics.textdb.planstore.PlanStoreConstants;
+import edu.uci.ics.textdb.web.TextdbWebException;
+import edu.uci.ics.textdb.web.request.QueryPlanRequest;
+import edu.uci.ics.textdb.web.request.beans.QueryPlanBean;
+import edu.uci.ics.textdb.web.response.QueryPlanResponse;
+import edu.uci.ics.textdb.web.response.SampleResponse;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * Created by kishorenarendran on 2/24/17.
+ */
+@Path("/planstore")
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+public class PlanStoreResource {
+
+    private static PlanStore planStore;
+    private static ObjectMapper mapper;
+
+    static {
+        try {
+            planStore = PlanStore.getInstance();
+            mapper = new ObjectMapper();
+            mapper.configure(MapperFeature.USE_GETTERS_AS_SETTERS, false);
+            mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+            planStore.createPlanStore();
+        }
+        catch (TextDBException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @GET
+    public QueryPlanResponse getAllQueryPlans() {
+        ArrayList<QueryPlanBean> queryPlans = new ArrayList<>();
+
+        try {
+            // Getting an iterator for the plan store
+            IDataReader reader = planStore.getPlanIterator();
+            reader.open();
+
+            // Iterating through the stored plans, and mapping them to a QueryPlanRequest object
+            ITuple tuple;
+            while ((tuple = reader.getNextTuple()) != null) {
+                String name = tuple.getField(PlanStoreConstants.NAME).getValue().toString();
+                String description = tuple.getField(PlanStoreConstants.DESCRIPTION).getValue().toString();
+                String logicalPlanJson = tuple.getField(PlanStoreConstants.LOGICAL_PLAN_JSON).getValue().toString();
+                queryPlans.add(new QueryPlanBean(name, description,
+                        mapper.readValue(logicalPlanJson, QueryPlanRequest.class)));
+            }
+        }
+        catch(TextDBException e) {
+            throw new TextdbWebException(e.getMessage());
+        }
+        catch(IOException e) {
+            e.printStackTrace();
+        }
+        QueryPlanResponse queryPlanResponse = new QueryPlanResponse(queryPlans);
+        return queryPlanResponse;
+    }
+
+    @GET
+    @Path("/{plan_name}")
+    public QueryPlanBean getQueryPlan(@PathParam("plan_name") String planName) {
```

Comment:
In `QueryPlanResource.executeQueryPlan`, we return a response and manually constructs the response object: `Response.status(200).entity(...).build()`, or in case of an error `Response.status(400).entity(...).build()`.

Can we change it to something similar to this? Which way is considered better?